src/s3path.jl

```diff
@@ -1,4 +1,6 @@
-struct S3Path{A<:AbstractAWSConfig} <: AbstractPath
+const S3PathConfig = Union{AbstractAWSConfig, Nothing}
```

Comment:
Should probably be located with `AbstractS3Version` or `AbstractS3Version` should be re-located here. Adding the "Abstract" prefix to be consistent with `AbstractS3Version`
```suggestion
const AbstractS3PathConfig = Union{AbstractAWSConfig, Nothing}
```